cookbooks/creating_a_behat_extension.rst

```diff
@@ -0,0 +1,278 @@
+           * turn. If you want to initialise multiple contexts, you can of course give them an
+           * interface and check for that here.
+           */
+          if (!$context instanceof HelloWorldContext) {
+              return;
+          }
+
+          $context->initializeConfig($this->enable, $this->text);
+      }
+  }
+
+We need to register the initializer definition within the Behat container through the ``HelloWorldExtension``, ensuring it gets loaded:
+
+.. code-block:: php
+
+  <?php
+
+  // ...
+
+  use Symfony\Component\DependencyInjection\Definition;
+  use Behat\Behat\Context\ServiceContainer\ContextExtension;
+
+  class HelloWorldExtension implements Extension
+  {
+      // ...
+
+      public function load(ContainerBuilder $container, array $config)
+      {
+          $definition = new Definition(HelloWorldInitializer::class, [
+              $config['text'],
+              $config['enable'],
+          ]);
+          $definition->addTag(ContextExtension::INITIALIZER_TAG);
+          $container->setDefinition('helloworld_extension.context_initializer', $definition);
+      }
+
+      // ...
+  }
+
+To complete the extension, we must add methods to ``HelloWorldContext`` to receive the configuration values and use those in the hooks:
+
+.. code-block:: php
+
+  // ...
+
+  class HelloWorldContext implements Context
+  {
+      private bool $enable = false;
+      private string $text;
+
+      public function initializeConfig(bool $enable, string $text)
+      {
+          $this->enable = $enable;
+          $this->text = $text;
+      }
+
+      /** @Given I say Hello World */
+      public function helloWorld()
+      {
+          if ($this->enable === false) {
+              return;
+          }
+  
+          throw new PendingException($this->text);
+      }
+  }
+
+Using the extension
+-------------------
+
+Now that the extension is ready and will inject values into context, we just need to configure it into a project.
+
+In the ``extensions`` key of a profile (``default`` in our case), we'll add the ``HelloWorldExtension`` key and configure our ``text`` and ``enable`` value.
+
+Finally, we need to load the ``HelloWorld\Context\HelloWorldContext`` into our suite.
+
+Here's the ``behat.yaml``:
+
+.. code-block:: yaml
+
+  default:
+    suites:
+      default:
+        contexts:
+          - FeatureContext
+          - HelloWorld\Context\HelloWorldContext
+    extensions:
+      HelloWorld\ServiceContainer\HelloWorldExtension:
+        text: 'Hi there!'
+        enable: true
+
+And now a scenario like this one:
+
+.. code-block::
+  Feature: Test
+  
+    Scenario: Test
+      Given I say Hello World
+
+Will display our text as a pending text.
+
+Conclusion
+----------
+
+Congratulations! You have just created a simple Behat extension.
+This extension demonstrates the three essential steps to building a Behat extension: defining an extension, creating an initializer, and configuring contexts.
```

Comment:
"essential" isn't quite right here as a lot of extensions don't need to create initializers or configure contexts.

```suggestion
This extension demonstrates three of the common steps to building a Behat extension: defining an extension, creating an initializer, and configuring contexts.
```